src/main/java/org/threadly/concurrent/PrioritySchedulerStatisticTracker.java

```diff
@@ -169,19 +129,11 @@ public PrioritySchedulerStatisticTracker(int corePoolSize, int maxPoolSize,
   public PrioritySchedulerStatisticTracker(int corePoolSize, int maxPoolSize,
                                            long keepAliveTimeInMs, TaskPriority defaultPriority, 
                                            long maxWaitForLowPriorityInMs, ThreadFactory threadFactory) {
-    super(corePoolSize, maxPoolSize, keepAliveTimeInMs, 
-          defaultPriority, maxWaitForLowPriorityInMs, 
-          threadFactory);
+    super(new StatisticWorkerPool(threadFactory, corePoolSize, maxPoolSize, 
+                                  keepAliveTimeInMs, maxWaitForLowPriorityInMs, new StatsManager()), 
+          defaultPriority);
     
-    totalHighPriorityExecutions = new AtomicInteger();
-    totalLowPriorityExecutions = new AtomicInteger();
-    runningTasks = new ConcurrentHashMap<Wrapper, Long>();
-    int endPadding = MAX_WINDOW_SIZE * 2;
-    runTimes = new ConcurrentArrayList<Long>(0, endPadding);
-    lowPriorityWorkerAvailable = new ConcurrentArrayList<Boolean>(0, endPadding);
-    highPriorityWorkerAvailable = new ConcurrentArrayList<Boolean>(0, endPadding);
-    lowPriorityExecutionDelay = new ConcurrentArrayList<Long>(0, endPadding);
-    highPriorityExecutionDelay = new ConcurrentArrayList<Long>(0, endPadding);
+    this.statsManager = ((StatisticWorkerPool)workerPool).statsManager;
```

Comment:
We must provide our version of the "WorkerPool" into the parent class so we can extend it's functionality.  Because we also need to have a handle on our implementation, and because we can't reference this before calling the super constructor, we have to cast it here.

I hate casting, but I see no other option.  We are have the same thing happening in ControlledThreadProfiler.